internal/provider/waypoint/data_source_waypoint_application_template.go

```diff
@@ -123,6 +125,31 @@ func (d *DataSourceApplicationTemplate) Schema(ctx context.Context, req datasour
 					},
 				},
 			},
+			"variable_options": schema.ListNestedAttribute{
+				Computed:    true,
+				Description: "List of variable options for the template",
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"name": &schema.StringAttribute{
+							Required:    true,
+							Description: "Variable name",
+						},
+						"variable_type": &schema.StringAttribute{
+							Required:    true,
+							Description: "Variable type",
+						},
+						"options": &schema.ListAttribute{
+							ElementType: types.StringType,
+							Required:    true,
+							Description: "List of options",
+						},
+						"user_editable": &schema.BoolAttribute{
+							Required:    true,
```

Comment:
nit/fyi: these `Required` fields don't make sense because this is a data-source. Don't feel like you need to change it though if TF doesn't complain